src/libos/src/fs/events.rs

```diff
@@ -38,6 +40,32 @@ impl IoEvents {
 
 impl Event for IoEvents {}
 
+pub trait AtomicIoEvents {
+    /// Update the IoEvents atomically.
+    ///
+    /// The update is equivalent to the following assignment
+    /// ```
+    /// self.store(self.load(Ordering::Relaxed) & !*mask | *ready, ordering)
+    /// ```
+    fn update(&self, ready: &IoEvents, mask: &IoEvents, ordering: Ordering);
```

Comment:
Do not get the point why do we need to give user the flexibility to change the ordering?